refactor: display account group number in connection list item

montelaidev opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR updates the `ConnectionListItem` to show account groups in state two.

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/35427?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: Display the number of account group instead of addresses for a connected dapp. 

## **Related issues**

Related to: https://consensyssoftware.atlassian.net/browse/MUL-669?atlOrigin=eyJpIjoiNDk4MmQ0YTFiMDg0NDFkMzkzMjE0NTQxOTJlYTk2NjAiLCJwIjoiaiJ9

## **Manual testing steps**

With state 2 enabled for multichain accounts. 

1. Connect to a dapp and open the wallet
2. Go to settings then permissions
3. See that it is displaying the number of account groups instead of number of accounts

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **After**

<img width="400" height="596" alt="Screenshot 2025-08-27 at 18 08 47" src="https://github.com/user-attachments/assets/fccfc14c-b2eb-4491-8b17-cabb36f19e55" />

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

montelaidev committed:
feat: update connection list item to show account groups in state two.

montelaidev committed:
refactor: use set

montelaidev committed:
fix: test

montelaidev committed:
fix: storybook

owencraston commented on ui/components/multichain/pages/permissions-page/connection-list-item.js:
```diff
@@ -25,14 +25,39 @@ import {
+
+  const isState2Enabled = useSelector(getIsMultichainAccountsState2Enabled);
+
+  const acccountsToShow = useMemo(() => {
```
typo. acccountsToShow -> accountsToShow

owencraston commented on ui/components/multichain/pages/permissions-page/connection-list-item.js:
```diff
@@ -25,14 +25,39 @@ import {
+      accountGroups.forEach((group) => {
+        group.accounts.forEach((account) => {
+          accountAddressesSet.add(account.address);
+        });
```
something like this...

```
const accountAddressSet = useMemo(() => {
  if (!isState2Enabled) {
    return null;
  }
  const set = new Set();
  (accountGroups ?? []).forEach((group) => {
    (group.accounts ?? []).forEach((account) => {
      set.add(account.address);
    });
  });
  return set;
}, [isState2Enabled, accountGroups]);


const accountsToShow = useMemo(() => {
  if (!isState2Enabled) {
    return connection.addresses?.length ?? 0;
  }
  if (!accountAddressSet || !connection.addresses?.length) {
    return 0;
  }
  let count = 0;
  for (const address of connection.addresses) {
    if (accountAddressSet.has(address)) {
      count++;
    }
  }
  return count;
}, [isState2Enabled, accountAddressSet, connection.addresses]);
```

montelaidev committed:
feat: add support for singular account display in connection list item

- Introduced a new message key for "account" in the localization file.
- Updated the logic in the ConnectionListItem component to correctly display the singular form of "account" when only one account is present.
- Adjusted the snapshot tests to reflect the changes in account display.

montelaidev committed:
Merge branch 'main' into refactor/connection-list-item-account-group

montelaidev committed:
fix: add locale to en_gb

montelaidev committed:
fix: snapshot

vinnyhoward reviewed (approved):
LGTM
